### ISSUE TITLE
Ability to define <IntlProvider> at the root component and then access it anywhere among its children.

### ISSUE THREAD

**Issue by piyaa** [yahoo/react-intl#717]
Is there a way to define <IntlProvider></IntlProvider> at the root comp, somewhat like this:

render() {
  return (
    <IntlProvider locale="ja">
     <App>
         <Child></Child>
         <Child2></Child2>
         <Child3></Child3>
    </App>
    </IntlProvider>
  )
}

now if I try to use the "formatteddate" in <Child3> comp, it does not work as expected, but when I add the add the <IntlProvider> inside that comp it works, can we not have something that defines the <IntlProvider> at the top comp and use it anywhere in the app, w/o having to re-define the<IntlProvider> again in the child comps?

**Comment by ericf** [comment]
What version of React are you using? This should work as expected, and as shown in all the `examples/` in this repo with React 14+

**Comment by piyaa** [comment]
Im using React 15 and i tried this:

below os my root comp where i'd like to add the "IntlProvider".
export default class PortalModule {
  render() {
     return (
      <IntlProvider locale="ja">
       <App/>
     </IntlProvider>
    )
  }
}


export default class App extends React.Component {
  render() {
   <Child1>
   <div style= {paddingTop:"135px"}> {this.props.children}</div>
   <Child1/>
  }
}

export default class Child1 extends React.Component {
    render() {
   <Child2>
   <div style= {paddingTop:"135px"}> {this.props.children}</div>
   <Child2/>
  }
}

export default class Child2 extends React.Component {
    render() {
       <div> <FormattedDate
                    value={new Date()}
                    day="numeric"
                    month="short"
                    year="numeric" />
          </div>
        <div style= {paddingTop:"135px"}> {someContent}</div>
  }
}

Basically I have nested components and I'm trying to access the 3rd/4th child of the root comp, with this i dont see any changes happening, what worng im doing?
